webapp/components/reftest-analyzer.js

```diff
@@ -157,75 +167,93 @@ class ReftestAnalyzer extends PolymerElement {
 
   async setupZoomSVG() {
     const zoomed = this.shadowRoot.querySelector('#zoomed');
-    const paths = [];
-    for (let x = 0; x < 5; x++) {
-      paths.push([]);
-      for (let y = 0; y < 5; y++) {
-        const path = document.createElementNS(nsSVG, 'path');
-        const offsetX = x * 50 + 1;
-        const offsetY = y * 50 + 1;
-        path.setAttribute('d', `M${offsetX},${offsetY} H${offsetX + 48} V${offsetY + 48} H${offsetX} V${offsetY}`);
-        path.setAttribute('fill', blankFill);
-        paths[x].push(zoomed.appendChild(path));
+    const pathsBefore = [], pathsAfter = [];
+    for (const before of [true, false]) {
+      const paths = before ? pathsBefore : pathsAfter;
+      for (let x = 0; x < 5; x++) {
+        paths.push([]);
+        for (let y = 0; y < 5; y++) {
+          const path = document.createElementNS(nsSVG, 'path');
+          const offsetX = x * 50 + 1;
+          const offsetY = y * 50 + 1;
+          if (before) {
+            path.setAttribute('d', `M${offsetX},${offsetY} H${offsetX + 48} L${offsetX},${offsetY + 48} V${offsetY}`);
+          } else {
+            path.setAttribute('d', `M${offsetX + 48},${offsetY} V${offsetY + 48} H${offsetX} L${offsetX + 48},${offsetY}`);
+          }
+          path.setAttribute('fill', blankFill);
+          paths[x].push(zoomed.appendChild(path));
+        }
       }
     }
-    this.paths = paths;
+    this.pathsBefore = pathsBefore;
+    this.pathsAfter = pathsAfter;
   }
 
-  async computeDiff(canvasBefore, canvasAfter) {
+  computeDiff(canvasBefore, canvasAfter) {
     if (!canvasBefore || !canvasAfter) {
       return;
     }
-    const before = this.shadowRoot.querySelector('#before');
-    const after = this.shadowRoot.querySelector('#after');
-
-    const beforeCtx = canvasBefore.getContext('2d');
-    const afterCtx = canvasAfter.getContext('2d');
-
-    const out = document.createElement('canvas');
-    out.width = Math.max(before.width, after.width);
-    out.height = Math.max(before.height, after.height);
-    const outCtx = out.getContext('2d');
-
-    for (let y = 0; y < Math.min(before.height, after.height); y++) {
-      const beforePixels = beforeCtx.getImageData(0, y, before.width, 1).data;
-      const afterPixels = afterCtx.getImageData(0, y, after.width, 1).data;
-      for (let x = 0; x < Math.min(before.width, after.width); x++) {
-        for (let i = 0; i < 4; i++) {
-          const pxlBefore = beforePixels[(x * 4) + i];
-          const pxlAfter = afterPixels[(x * 4) + i];
-          if (pxlBefore !== pxlAfter) {
-            outCtx.fillRect(x, y, 1, 1);
-            break;
+    return this.load(new Promise(resolve => {
```

Comment:
This code is unchanged except for the `this.load` so that we show a spinner while it's computing. On my machine, this only shows for like, 100ms. On slower machine it may take longer.